packages/nextjs/app/api/users/register/route.ts

```diff
@@ -36,31 +36,33 @@ export async function POST(req: Request) {
       userAddress: address,
     };
 
-    let ensName: string | null = null;
-    try {
-      ensName = await publicClient.getEnsName({ address });
+    const user = await createUser(userToCreate);
 
-      if (ensName) {
-        userToCreate.ens = ensName;
-      }
-    } catch (error) {
-      console.error(`Error getting ENS name for user ${address}:`, error);
-    }
+    // Background processing
+    waitUntil(
+      (async () => {
+        try {
+          await trackPlausibleEvent(PlausibleEvent.SIGNUP_SRE, {}, req);
 
-    if (ensName) {
-      try {
-        const ensAvatar = await publicClient.getEnsAvatar({ name: ensName });
-        if (ensAvatar) {
-          userToCreate.ensAvatar = ensAvatar;
-        }
-      } catch (error) {
-        console.error(`Error getting ENS avatar for user ${address}:`, error);
-      }
-    }
+          const { ensData } = await fetchOnchainData(address);
 
```

Comment:
Shall we split this into two `waitUntil` one for analytics and one for fetching onchain data? 

Because currently if the `trackPlausibleEvent` fails it will not run the `fetchOnchainData`